public/app/plugins/datasource/prometheus/partials/config.html

```diff
@@ -1,3 +1,14 @@
 <datasource-http-settings current="ctrl.current" suggest-url="http://localhost:9090">
 </datasource-http-settings>
 
+<h3 class="page-heading">Prometheus details</h3>
+
+<div class="gf-form-group max-width-30">
+	<div class="gf-form">
+		<label class="gf-form-label width-13">Label Keys</label>
+		<input type="text" class="gf-form-input max-width-18" ng-model='ctrl.current.jsonData.labelKeys' placeholder="job,instance"></input>
```

Comment:
If it is OK to show all label keys in UI, no need to metric name.

If user start to edit adhoc filter, the dashboard might have some graph panel, and datasource plugin already receive query response.
Gathering label keys in the response, and provide label keys to adhoc filter UI is good choice, I think.
The label keys in query response is mostly usable in the dashboard.